### PR TITLE
Checkout Properties Refactor

### PR DESCRIPTION
**Pull request by tobitech**
# 📲 What

Refactor checkoutProperties into SharedFunctions

# 🤔 Why

We want to be able to reuse a function that creates checkout properties for different view models in the pledge flow

### CODE REVIEW

**Review by moderateepheezy — APPROVED** [review]

File: Library/SharedFunctions.swift [hunk 1]
```diff
@@ -388,3 +388,94 @@ public func isEndDateAfterToday(for reward: Reward) -> Bool {
...
+}
+
+public func checkoutProperties(
+  from project: Project,
+  baseReward: Reward,
+  rewards: [Reward],
+  selectedQuantities: SelectedRewardQuantities,
+  additionalPledgeAmount: Double,
+  pledgeTotal: Double,
+  shippingTotal: Double,
```
Comment (singhhari) [inline Library/SharedFunctions.swift:407]: Same here? If possible, we should try to make the function signature as simple as functionally possible. That way, whenever we need to use it we only need to pass what's absolutely necessary and all the logic required to create `CheckoutPropertiesData` is encapsulated in the method body.

File: Library/SharedFunctions.swift [hunk 2]
```diff
@@ -388,3 +388,94 @@ public func isEndDateAfterToday(for reward: Reward) -> Bool {
...
+ */
+public func rounded(_ value: Double, places: Int) -> Double {
+  let divisor = pow(10.0, Double(places))
+  return (value * divisor).rounded() / divisor
+}
+
+public func checkoutProperties(
+  from project: Project,
+  baseReward: Reward,
+  rewards: [Reward],
```
Comment (singhhari) [inline Library/SharedFunctions.swift:403]: I think rewards are a property of `project` or `project.rewardData` so I would double check if this parameter is necessary.

**Review by singhhari — CHANGES_REQUESTED** [review]
Added some comments that will require you to test this/ play around with it a bit. Try and simplify the method even more so we can use this more flexibly.

File: Library/SharedFunctions.swift (hunk 2 above)
Comment (tobitech) [reply Library/SharedFunctions.swift:403]: I think it represents selected rewards not all the project rewards, but I'm not so sure about that as well. I've seen cases in the project where a single selected reward was turned into an array of one item.

**Review by singhhari — APPROVED** [review]
lgtm. 

@moderateepheezy @tobitech let's try and be very observant of re-factoring further as we start using the method for more of the event work.

**Pull request merged**